packages/terra-clinical-embedded-content-consumer/src/EmbeddedContentConsumer.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Consumer } from 'xfc';
+import styles from './EmbeddedContentConsumer.scss';
+
+const propTypes = {
+  /**
+   * The source URL of the content to load.
+   */
+  src: PropTypes.string.isRequired,
+  /**
+  * A set of event handlers keyed by the event name.
+  */
+  eventHandlers: PropTypes.arrayOf(PropTypes.shape({
+    key: PropTypes.string,
+    handler: PropTypes.func,
+  })),
+};
+
+class EmbeddedContentConsumer extends React.Component {
+
+  componentDidMount() {
+    Consumer.init();
+    this.xfcFrame = Consumer.mount(this.embeddedContentWrapper, this.props.src);
+
+    if (this.props.eventHandlers) {
+      this.props.eventHandlers.forEach((event) => {
```

Comment:
Also, XFC uses event emitter so this lets you do the following
* Use regex to catch events 
* Pass a single object and bind all events at once

https://github.com/Olical/EventEmitter/blob/master/docs/api.md#addlisteners